### PR TITLE
wip: alpha release

### PR DESCRIPTION
**Pull request by bitwalker**
This PR supercedes #260, and introduces a `next` branch on which our usual development will take place once merged.

@greenhat This is the same changes as #260, just rebased on `main` to pick up the few PRs of yours we merged already. I'd fork from `next` for all subsequent PRs.

Once this is merged, it's because the requirements for the alpha release have been met (i.e. the memory init issues have been worked out, and CI is green), hence the naming of this PR.

Until merged, stack PRs on top of this one, and we can follow our usual process that way.

### CODE REVIEW

**Review by greenhat — APPROVED** [review]

**Comment by greenhat** [comment]
We need to make sure this branch will not be deleted after the merge.

**Comment by greenhat** [comment]
@bitwalker `build_new_project_from_template` test fails with `Failed to compile:   x linker error: undefined module '#miden::account'` because `miden::account` is the SDK module for tx kernel functions. I'm looking into setting `Linker::allow_missing`. I'm thinking of hard-coding it with tx kernel module names. What do you think?

**Comment by bitwalker** [comment]
I'm planning to merge this later today once CI is green and I've looked at any other outstanding PRs we have. After that we can resume our usual development flow with `next` as our main development branch, which we've kind of been doing, but it is a bit awkward at the moment with `next` not being a protected branch.

**Pull request merged**
